src/3p-frame-messaging.js

```diff
@@ -120,9 +121,11 @@ export function deserializeMessage(message) {
   }
   const startPos = message.indexOf('{');
   devAssert(startPos != -1, 'JSON missing in %s', message);
-  return tryParseJson(message.substr(startPos), (e) =>
-    dev().error('MESSAGING', 'Failed to parse message: ' + message, e)
-  );
+  return tryParseJson(message.substr(startPos), (e) => {
+    rethrowAsync(
+      new Error('MESSAGING: Failed to parse message: ' + message, e.message)
```

Comment:
The `e.message` will be dropped because it's being passed as the second argument to the `Error` constructor:

```suggestion
      new Error(`MESSAGING: Failed to parse message: ${message}\n${e.message}`)
```